src/script/cpp_api/s_security.cpp

```diff
@@ -275,66 +300,57 @@ void ScriptApiSecurity::initializeSecurityClient()
 
 	lua_State *L = getStack();
 
-
-	int old_globals = backupGlobals(L);
-
+	// Replace the global environment with a new one
+#if LUA_VERSION_NUM <= 501
+	int is_main = lua_pushthread(L);  // Push the main thread
+	FATAL_ERROR_IF(!is_main, "Security: ScriptApi's Lua state "
+		"isn't the main Lua thread!");
+#endif
+	lua_newtable(L);  // Create new environment
+	lua_pushvalue(L, -1);
+	lua_setfield(L, -2, "_G");  // Set _G of new environment
```

Comment:
For this, I think you should add more helper methods to remove the duplication created by removing `backupGlobals`.
Here's the general idea:
```C++
{
  //...
  int new_env = pushEmptyEnv(L);
  int old_env = lua_getglobal(L, "_G");
  // Copy from old_env to new_env
  lua_pop(L, 1); // Pop old_env
  setGlobalEnv(L, new_env);
}
void loadGlobalEnv(lua_State *L, int idx) {
  // Lua 5.1/5.2 environment setting code, pops new env
}
int pushEmptyEnv(lus_State *L) {
  // Pushes table with self-referential _G
  return lua_gettop(L);
}
```